### ISSUE TITLE
Parallel tests for Firefox running all on one of Docker instances

### ISSUE THREAD

**Issue by mbranicky** [cypress-io/cypress#9090]
### Current behavior
After update to Cypress 5.4.0 and docker image `cypress/browsers:node12.14.1-chrome85-ff81` we have noticed that when tests running in parallel they are distributed across 5 machines, as configured in `bitbucket-pipelines.yml`, but in one Docker instance the Cypress runs ALL tests. 

<img width="250" alt="cypress_firefox_pr" src="https://user-images.githubusercontent.com/5552744/98115846-ba5d9900-1ea7-11eb-976e-b137bca7f5b6.png">

This happens only **for Firefox**, not for Chrome. Both using the same Docker image.

Tests are executed as:
```
cypress run --config-file cypress-no-smoke.json "--browser" "firefox" "--record" "--group" "5 x Firefox 81" "--parallel" "--ci-build-id" "2274"
```

Of course the `ci-build-id` is set from `$BITBUCKET_BUILD_NUMBER` variable. This is causing significantly longer runs of tests as well as it counts against our Cypress Dashboard Service quota.

### Desired behavior

All Firefox tests are distributed across parallel machines as it works in Cypress 5.3.0.

### Versions
5.3.0. was working
5.4.0
5.5.0

Repository is private I cannot provide details, but CI provider is Bitbucket. 
Probably related to: 
https://github.com/cypress-io/cypress/issues/8912#issue-726678061

Please let me know which additional details should I provide.

**Comment by mbranicky** [comment]
I've just noticed that at the beginning of every test run Cypress informs about current "_Run URL_", like https://dashboard.cypress.io/projects/XYZ/runs/791.

From some reason Cypress assigns to that long running Firefox test instance the same run number (i.e. 791) as to the previous Chrome instances (all have 791). For the rest of Firefox instances they got next available run number, in this case 792.

So in Dashboard there are now two different runs for the same build.
First run (791) is a complete set of Chrome machines plus that Firefox one which executes all tests and takes too long.
And second run (792) which contains only remaining 4 Firefox instances.

I want to point out that we did not make any Dashboard nor Bitbucket configuration changes, except of upgrading to Cypress 5.4.0.

**Comment by jennifer-shehane** [comment]
Yes, we did make a change concerning Bitbucket specifically in 5.4.0 that is likely effecting your runs in this unintended way. https://github.com/cypress-io/cypress/issues/8720

We changed some of the logic so that 'reruns' are recognized as separate runs. It's often difficult for us to determine a rerun from the original run because many CI providers send the exact same ID for both (like the `BITBUCKET_BUILD_NUMBER`), so we take this ci build id and append a `-2` to the end of it when we determine it is a 'rerun'. 

Likely, you would see that your 791 run has the cibuildid you passed, but the 792 run has `cibuildid-2`, with a `-2` appended on the end of it. 

You can see the evaluated `cibuildid` in the Dashboard in the area shown below:

<img width="1348" alt="Screen Shot 2020-11-05 at 12 05 45 PM copy" src="https://user-images.githubusercontent.com/1271364/98202042-74fca280-1f5f-11eb-9259-17687b370929.png">

This is not the correct behavior when the specs are run within the same pipeline (not a rerun) with the same `cibuildid`.

In the PR from 5.4.0, I can see that we're collecting the `BITBUCKET_PARALLEL_STEP` environment variable to help us in determining if it is a unique job (so we can distinguish a rerun). 

**Followup Questions**

- Are you overwriting the `BITBUCKET_PARALLEL_STEP` value anywhere? 
- Did anything within this particular run involve 'rerunning' the pipeline? 

I'll have our API team take a look into this on our side.

**Comment by mbranicky** [comment]
To answer your questions first:

- we are not intentionally overwriting this variable at all, unless would be done by Bitbucket, which is not very likely. In fact as a step before tests, we run the `npx @bahmutov/print-env BITBUCKET CYPRESS` to list all Bitbucket and Cypress related ENV variables. Thanks to this I've been able to collect some data for you in attached file.
- I've checked several past builds and I can confirm that re-run wasn't involved in any of these cases. In fact it would be clearly indicated in the Bitbucket Pipeline overview, similar as I've attached in my original post.

Regarding of the `BITBUCKET_PARALLEL_STEP` I wonder how it can help to detect the re-runs, cause the value of this variable is in fact the index of specific step within the single batch of parallel steps. Maybe for this purpose the combination of `BITBUCKET_BUILD_NUMBER` and `BITBUCKET_STEP_RUN_NUMBER` should be used. I think the second one indicates the number of how many times the specific step has had run. But I am not a Bitbucket expert.

I've tried to collect some data for you from one of our problematic builds, basically I chose several I think relevant `ENV` variables, for each step ( 5 x Chrome and 5 x Firefox). So you can take a look which are step specific and which are build/run specific: [bitbucket_steps_env_vars.txt](https://github.com/cypress-io/cypress/files/5493015/bitbucket_steps_env_vars.txt)

Please  let me know if you need additional data or details.

**Comment by jennifer-shehane** [comment]
This issue was fixed Friday, November 13 by a release from our API. The fix should be immediate and will not require an update to a newer version of Cypress. 

We reverted the previous changes, which were causing unintended consequences when anyone used multiple concurrent pipelines and will look to other solutions to solving the issue with rerunning Bitbucket pipelines.

Thanks for your patience!
